site/content/entry/citra-progress-report-2023-q1/index.md

```diff
@@ -0,0 +1,390 @@
+After the initial implementation by [nieldm](https://github.com/nieldm) came a slew of other issues, such as internal issues with the custom paths and every user being accidentally forced to use custom directories for SDMC and NAND!
+
+All of these issues were ironed out by the joint effort of our developers and we’re proud to say that custom directories for SDMC and NAND are here to stay!
+
+## Start abstracting the rasterizer cache from OpenGL ([#6013](https://github.com/citra-emu/citra/pull/6103)) by [GPUCode](https://github.com/GPUCode)
+
+As we are entering the mid 2023’s, Citra is long due a makeover!
+First on the list of things to make Citra shiny again is a polish of our OpenGL graphics backend.
+
+A fundamental part of Citra’s renderer, regardless of the graphics API used, is a component known as the rasterizer cache, which is responsible for tracking all memory reads and writes to the emulated 3DS VRAM. The rasterizer cache is Citra’s version of the texture cache, a concept well known to avid emulation followers. 
+
+*But why is this needed?* 
+To answer said question, we need to look closer into the architecture of the 3DS. It, alongside other Nintendo consoles of the era like the Wii, uses a unified memory architecture (UMA), allowing both the CPU and the GPU to access the same memory region. Your computer, on the other hand, works much differently; the system RAM is separate from the video RAM. Access to system RAM from the CPU is relatively fast, something that’s not the case when it comes to video RAM. The latter can also be accessed by the CPU. There’s one problem though; doing that is very **slow**.
+
+The main function of the rasterizer cache in Citra is to keep textures in the host VRAM, in the computer’s GPU, so that the renderer can access them quickly whilst also keeping the host VRAM in sync with the emulated 3DS VRAM. This is particularly necessary for resolution upscaling. As an example, if you had to upload a 2560x1400 texture each frame, the performance would be terrible!
+
+In the past, this rasterizer cache was tied with OpenGL and used OpenGL methods directly. This PR introduces a class called `TextureRuntime`. This is used to abstract away texture operations from the rasterizer cache. By doing this, it allows for other graphic APIs to be easily ported over to Citra, which we will get into more later in this report.
+
+## 3GX Plugin Loader ([#6172](https://github.com/citra-emu/citra/pull/6172)) by [PabloMK7](https://github.com/PabloMK7)
+
+`.3gx` plugins are executable files that are mapped into memory and executed when the process runs. This allows developers to expand the behavior of games in a way that would normally be impossible, such as an Action Replay code.
```

Comment:
Not a native speaker, but I think some context is missing here to explain what you mean with "the process".

Reply:
I actually have no idea what "the process" means. That was just copied exactly from the PR. I'll message Pablo about it and will update that one when they reply.